### PR TITLE
#43 - Add interfaces for system models of Shopware 6

### PR DESCRIPTION
**Pull request by rmakara**
**Changes:**
* Add interfaces for `system` group based on Shopware 6 data structures.
* These models seem to be required for covering Shopware API endpoints that we will need.
* It creates the following structure below `system`:
  * country,
  * currency,
  * locale, 
  * number-range,
  * sales-channel, 
  * salutation,
  * state-machine,
  * system-config,
  * tag,
  * tax,
  * unit,
  * user.

**Issue:**
#43

### CODE REVIEW

File: packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannel.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  countryId: string;
+  navigationCategoryId: string;
+  footerCategoryId: string | null;
+  serviceCategoryId: string | null;
+  name: string | null;
+  shortName: string | null;
+  accessKey: string;
+  currencies: CurrencyTranslation[] | null;
+  languages: Language[] | null;
+  configuration: [] | null;
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannel.ts]: missing type

File: packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannelType.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { CustomField } from "../../common/CustomField";
+
+export interface SalesChannelType {
+  name: string | null;
+  manufacturer: string | null;
+  description: string | null;
+  descriptionLong: string | null;
+  coverUrl: string | null;
+  iconName: string | null;
+  screenshotUrls: [] | null;
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannelType.ts]: missing type

File: packages/shopware-6-client/src/interfaces/models/system/state-machine/StateMachineHistory.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { StateMachineState } from "./StateMachineState";
+import { StateMachine } from "./StateMachine";
+import { User } from "../user/User";
+
+export interface StateMachineHistory {
+  stateMachineId: string;
+  stateMachine: StateMachine | null;
+  entityName: string;
+  entityId: [];
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/models/system/state-machine/StateMachineHistory.ts]: missing type

File: packages/shopware-6-client/src/interfaces/models/system/tag/Tag.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Product } from "../../content/product/Product";
+import { Media } from "../../content/media/Media";
+import { Category } from "packages/shopware-6-client/src";
+import { Order } from "../../checkout/order/Order";
+import { NewsletterRecipient } from "../../content/newsletter/NewsletterRecipient";
+import { ShippingMethod } from "../../checkout/shipping/ShippingMethod";
+
+export interface TagEntity {
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/models/system/tag/Tag.ts]:
```suggestion
export interface Tag {
```

File: packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannel.ts (hunk 1 above)
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannel.ts]: Also missing in Shopware. Just an array.

File: packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannelType.ts (hunk 2 above)
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/system/sales-channel/SalesChannelType.ts]: Just an array in Shopware.

File: packages/shopware-6-client/src/interfaces/models/system/state-machine/StateMachineHistory.ts (hunk 3 above)
Comment (rmakara) [reply packages/shopware-6-client/src/interfaces/models/system/state-machine/StateMachineHistory.ts]: Just an array in Shopware.

**Comment by rmakara** [comment]
@patzick types were missing also in Shopware. Just arrays for now. :( Name of TagEntity changed. :)

**Pull request merged**